main.go

```diff
@@ -62,60 +64,59 @@ type config struct {
 	flagset.StringSliceVar(&cfg.ignorePaths, "ignore-paths", nil, "Comma-separated list of paths against which kube-rbac-proxy pattern-matches the incoming request. If the requst matches, it will proxy the request without performing an authentication or authorization check. Cannot be used with --allow-paths.")
 
 	// TLS flags
-	flagset.StringVar(&cfg.tls.certFile, "tls-cert-file", "", "File containing the default x509 Certificate for HTTPS. (CA cert, if any, concatenated after server cert)")
-	flagset.StringVar(&cfg.tls.keyFile, "tls-private-key-file", "", "File containing the default x509 private key matching --tls-cert-file.")
-	flagset.StringVar(&cfg.tls.minVersion, "tls-min-version", "VersionTLS12", "Minimum TLS version supported. Value must match version names from https://golang.org/pkg/crypto/tls/#pkg-constants.")
-	flagset.StringSliceVar(&cfg.tls.cipherSuites, "tls-cipher-suites", nil, "Comma-separated list of cipher suites for the server. Values are from tls package constants (https://golang.org/pkg/crypto/tls/#pkg-constants). If omitted, the default Go cipher suites will be used")
+	flagset.StringVar(&cfg.ServingOptions.ServerCert.CertKey.CertFile, "tls-cert-file", "", "File containing the default x509 Certificate for HTTPS. (CA cert, if any, concatenated after server cert)")
+	flagset.StringVar(&cfg.ServingOptions.ServerCert.CertKey.KeyFile, "tls-private-key-file", "", "File containing the default x509 private key matching --tls-cert-file.")
+	flagset.StringVar(&cfg.ServingOptions.MinTLSVersion, "tls-min-version", "VersionTLS12", "Minimum TLS version supported. Value must match version names from https://golang.org/pkg/crypto/tls/#pkg-constants.")
+	flagset.StringSliceVar(&cfg.ServingOptions.CipherSuites, "tls-cipher-suites", nil, "Comma-separated list of cipher suites for the server. Values are from tls package constants (https://golang.org/pkg/crypto/tls/#pkg-constants). If omitted, the default Go cipher suites will be used")
 	flagset.DurationVar(&cfg.tls.reloadInterval, "tls-reload-interval", time.Minute, "The interval at which to watch for TLS certificate changes, by default set to 1 minute.")
 
 	// Auth flags
-	flagset.StringVar(&cfg.auth.Authentication.X509.ClientCAFile, "client-ca-file", "", "If set, any request presenting a client certificate signed by one of the authorities in the client-ca-file is authenticated with an identity corresponding to the CommonName of the client certificate.")
+	flagset.StringVar(&cfg.AuthenticationOptions.ClientCert.ClientCA, "client-ca-file", "", "If set, any request presenting a client certificate signed by one of the authorities in the client-ca-file is authenticated with an identity corresponding to the CommonName of the client certificate.")
+	flagset.BoolVar(&cfg.AuthenticationOptions.SkipInClusterLookup, "authentication-skip-lookup", true, ""+
```

Comment:
if the KIND cluster has a client CA generated for it and a functional CSR client API, then you can
1. set just this flag to --true, 
2. don't set the --client-ca-file
3. create a client cert/key pair
4. then use cluster-admin creds to approve the CSR
5. the kube-controller-manager should sign using a signing key that matches the in-cluster client-ca
6. get the signed cert from CSR status
7. then you can use your new client cert/key again the kube-rbac-server

Reply:
Done